### PR TITLE
moar languages

### PR DESCRIPTION
**Pull request by juli1**
## What problem are you trying to solve?

We need to add support for more languages: C#, Dockerfile, Go, Java, JSON, YAML

## What is your solution?

Add the necessary support in our build system

### CODE REVIEW

File: kernel/build.rs [hunk 1]
```diff
@@ -19,28 +19,96 @@ fn main() {
         name: String,       // the directory where we clone the project
         repository: String, // the repository to clone
         build_dir: PathBuf, // the directory we use to build the tree-sitter project
+        files: Vec<String>,
+        cpp: bool,
```
Comment (dastrong) [inline kernel/build.rs:23]: What's cpp?

File: kernel/src/model/common.rs [hunk 2]
```diff
@@ -41,10 +53,16 @@ static ALL_LANGUAGES: &[Language] = &[
...
+            Self::Csharp => "C#",
+            Self::Dockerfile => "Dockerfile",
+            Self::Go => "Go",
+            Self::Java => "java",
             Self::JavaScript => "javascript",
+            Self::Json => "json",
             Self::Python => "python",
             Self::Rust => "rust",
             Self::TypeScript => "typescript",
+            Self::Yaml => "yaml",
```
Comment (dastrong) [inline kernel/src/model/common.rs:65]: Some have capitals and other don't. Was there a reason for that?

**Review by dastrong — APPROVED** [review]

File: kernel/build.rs (hunk 1 above)
Comment (juli1) [reply kernel/build.rs]: It's C++ (shortcut for `CPlusPlus`)

File: kernel/src/model/common.rs (hunk 2 above)
Comment (juli1) [reply kernel/src/model/common.rs]: fixed!

**Pull request merged**